pkgs/servers/http/hiawatha/default.nix

```diff
@@ -7,7 +7,7 @@
 , libxslt
 , libxml2
 
-, enableSSL ? true
+, enableSystemTls ? true, mbedtls ? null
```

Comment:
We should always use system TLS, if we build with TLS support.

Reply:
---
Yes, as this is secure server - I enabled TLS by default.
Case when TLS would not be needed, but secure server needed - is possible, but probably not spread case.
---

System-wide thread-able dynamic library, - makes it a less secure variant to bundled library. 
Through threading process in some cases some information can be shared, or gathered.
https://tls.mbed.org/kb/development/thread-safety-and-multi-threading

That is why I made this switch. For truly paranoid people to crank-up the security more.

Sigh. Of course, everything always has some security explanation.

Reply:
I can imagine encryption to be disabled on embedded devices or local server in protected network.

Using bundled libraries is inherently less secure than using the system one. Nixpkgs maintainers can apply security fixes and update the library globally, long before hiawatha or other consumer updates. As for the link, I think it refers to memory safety (as usual with shared memory and concurrency), different applications would use different memory ranges.

Reply:
Yes. I also understood words as they are about parallel threads in one application can indirectly disclose some information to one another.

I would change comment notes.

As they are elementary, I implemented all Hiawatha build switches at this point. That looks like not too superfluous.

Reply:
If this is the case, it really would not matter when system library is used. The threads will use the same library in either case.